ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -137,13 +152,26 @@ public void setPlan(QueryPlan plan) {
       this.runtimeContext = Utilities.getFirstTezTask(plan.getRootTasks())
           .map(TezTask::getRuntimeContext)
           .orElse(null);
+      this.queryProperties = plan.getQueryProperties();
     }
   }
 
+  public String getQueryType() {
```

Comment:
i think those method do not belong here, and maybe `QueryInfo` would be a. better place
cc @kasakrisz

Reply:
queryType, getQueryType finally went to QueryProperties which already gathers useful information in easily understandable fields